yt_dlp/extractor/bilibili.py

```diff
@@ -502,39 +503,146 @@ def _real_extract(self, url):
             season_info.get('bangumi_title'), season_info.get('evaluate'))
 
 
-class BilibiliChannelIE(InfoExtractor):
-    _VALID_URL = r'https?://space.bilibili\.com/(?P<id>\d+)'
-    _API_URL = "https://api.bilibili.com/x/space/arc/search?mid=%s&pn=%d&jsonp=jsonp"
+class BilibiliSpaceBaseIE(InfoExtractor):
+    def __init__(self, downloader=None):
+        InfoExtractor.__init__(self, downloader)
+        self.page_idx_start = 1
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+
+        metadata, pagedlist = self._extract_internal(playlist_id)
+
+        return self.playlist_result(pagedlist, playlist_id, metadata['title'])
+
+    def _extract_internal(self, playlist_id, **kwargs):
+        first_page = self._fetch_page(playlist_id, self.page_idx_start, **kwargs)
+        metadata = self._get_metadata(first_page, **kwargs)
+
+        pagedlist = InAdvancePagedList(
+            lambda idx: self._get_entries(self._fetch_page(playlist_id, idx, **kwargs) if idx else first_page, **kwargs),
+            metadata['page_count'], metadata['page_size'])
+        return metadata, pagedlist
+
+    def _fetch_page(self, playlist_id, page_idx, **kwargs):
+        raise NotImplementedError('This method must be implemented by subclasses')
+
+    def _get_metadata(self, page_data, **kwargs):
+        raise NotImplementedError('This method must be implemented by subclasses')
+
+    def _get_entries(self, page_data, **kwargs):
+        raise NotImplementedError('This method must be implemented by subclasses')
```

Comment:
I would merge _fetch_page and _get_metadata into one, but this is about the order i was talking about

```python
def _fetch_page():
   ....

def _get_metadata():
    ....
    
def _get_entries():
   ....
   
def _extract_internal():
   ....
```

Reply:
_fetch_page will be call multi times, while _get_metadata only once,  I would like to keep it